deepchem/data/tests/test_data_loader.py

```diff
@@ -12,156 +9,153 @@
-        tasks=tasks, smiles_field="smiles", featurizer=featurizer)
-
-    dataset = loader.featurize(input_file)
-
-    # Splits featurized samples into train/test
-    splitter = dc.splits.RandomSplitter()
-    train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(
-        dataset)
-    assert len(train_dataset) == 8
-    assert len(valid_dataset) == 1
-    assert len(test_dataset) == 1
-
-  def random_test_train_test_split(self):
-    """Test of singletask RF ECFP regression API."""
-    #splittype = "random"
-    model_params = {}
-    tasks = ["log-solubility"]
-    task_type = "regression"
-    task_types = {task: task_type for task in tasks}
-    input_file = os.path.join(self.current_dir,
-                              "../../models/tests/example.csv")
-    featurizer = dc.feat.CircularFingerprint(size=1024)
-    loader = dc.data.CSVLoader(
-        tasks=tasks, smiles_field="smiles", featurizer=featurizer)
-
-    dataset = loader.featurize(input_file)
-
-    # Splits featurized samples into train/test
-    splitter = dc.splits.RandomSplitter()
-    train_dataset, test_dataset = splitter.train_test_split(dataset)
-    assert len(train_dataset) == 8
-    assert len(test_dataset) == 2
-
-  def test_log_solubility_dataset(self):
-    """Test of loading for simple log-solubility dataset."""
-    current_dir = os.path.dirname(os.path.realpath(__file__))
-    input_file = "../../models/tests/example.csv"
-    input_file = os.path.join(current_dir, input_file)
-
-    tasks = ["log-solubility"]
-    smiles_field = "smiles"
-    loader = dc.data.CSVLoader(
-        tasks=tasks,
-        smiles_field="smiles",
-        featurizer=dc.feat.CircularFingerprint(size=1024))
-    dataset = loader.featurize(input_file)
-
-    assert len(dataset) == 10
-
-  def test_dataset_move(self):
-    """Test that dataset can be moved and reloaded."""
-    base_dir = tempfile.mkdtemp()
-    data_dir = os.path.join(base_dir, "data")
-    moved_data_dir = os.path.join(base_dir, "moved_data")
-    dataset_file = os.path.join(self.current_dir,
-                                "../../models/tests/example.csv")
-
-    featurizer = dc.feat.CircularFingerprint(size=1024)
-    tasks = ["log-solubility"]
-    loader = dc.data.CSVLoader(
-        tasks=tasks, smiles_field="smiles", featurizer=featurizer)
-    featurized_dataset = loader.featurize(dataset_file, data_dir)
-    n_dataset = len(featurized_dataset)
-
-    # Now perform move
-    shutil.move(data_dir, moved_data_dir)
-
-    moved_featurized_dataset = dc.data.DiskDataset(moved_data_dir)
-
-    assert len(moved_featurized_dataset) == n_dataset
+def test_unlabelled():
+  current_dir = os.path.dirname(os.path.abspath(__file__))
+  input_file = os.path.join(current_dir, "../../data/tests/no_labels.csv")
+  featurizer = dc.feat.CircularFingerprint(size=1024)
+  loader = dc.data.CSVLoader(
+      tasks=[], smiles_field="smiles", featurizer=featurizer)
+  loader.create_dataset(input_file)
```

Comment:
Maybe add another `assert len(X)` here to check the dataset creation?

Reply:
Good suggestion, will add!